x-pack/plugins/data_visualizer/public/application/common/components/results_links/results_links.tsx

```diff
@@ -93,22 +106,36 @@ export const ResultsLinks: FC<Props> = ({
 
     getDiscoverUrl();
 
-    Promise.all(
-      additionalLinks.map(async ({ canDisplay, getUrl }) => {
-        if ((await canDisplay({ indexPatternId: dataViewId })) === false) {
-          return null;
-        }
-        return getUrl({ globalState, indexPatternId: dataViewId });
-      })
-    ).then((urls) => {
-      const linksById = urls.reduce((acc, url, i) => {
-        if (url !== null) {
-          acc[additionalLinks[i].id] = url;
-        }
-        return acc;
-      }, {} as Record<string, string>);
-      setGeneratedLinks(linksById);
-    });
+    if (Array.isArray(asyncLinkCards)) {
+      Promise.all(
+        asyncLinkCards.map(async (asyncCardGetter) => {
+          const results = await asyncCardGetter({
+            dataViewId,
+          });
+          if (Array.isArray(results)) {
+            return await Promise.all(
+              results.map(async (c) => ({
+                ...c,
+                canDisplay: await c.canDisplay(),
+                href: await c.getUrl(),
+              }))
+            );
+          }
+
+          if (results) {
```

Comment:
as mentioned above, if we only ever dealt with `ResultLink` arrays, this logic could be removed

Reply:
alternatively, if we do still need single result links, the single link could be put in an array and processed the same, e.g.

```ts
const tempResults = Array.isArray(results) ? results : [results];
return await Promise.all(results.map ....
```

Reply:
Updated here https://github.com/elastic/kibana/pull/131342/commits/846bd87fbe77a488d47e1a9ee7d2d56a17011e74